tests/SlidingPaginationSubscriberTest.php

```diff
@@ -91,6 +92,27 @@ public function testRemoveDefaultSortParamsNotIsset(): void
         ], $paginationParams);
     }
 
+    public function testPageLimitSet()
+    {
+        $this->subscriberOptions['defaultPageLimit'] = 50;
+
+        $paginationEvent = new Event\PaginationEvent;
+        $paginationEvent->options = &$this->options;
+
+        $slidingPaginationSubscriber = new SlidingPaginationSubscriber($this->subscriberOptions);
+        $slidingPaginationSubscriber->pagination($paginationEvent);
+        $pagination = $paginationEvent->getPagination();
+
+        $pagination->setItemNumberPerPage(1);
```

Comment:
An alternative would be to add a public `getPageLimit` method to the `SlidingPaginator` and checking that here. Would make this test a bit shorter. I'm not sure which is preferred.